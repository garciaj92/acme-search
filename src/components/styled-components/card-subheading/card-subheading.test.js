import React from "react";
import CardSubHeading from "./card-subheading";

import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';

// Mocks
import { consoleErrorFn } from "../../../__mocks__/console.mock";

const props = {
    heading: "Test Heading"
};

describe("CardSubHeading: Test visual of component", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CardSubHeading {...props} />);
    });

    it("renders correctly", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("contains correct href and content", () => {
        expect(wrapper.find("h5").first().text()).toEqual(props.heading);
    });
});

describe("CardSubHeading: Test prop type validation", () => {
    let consoleErrorSpy;
    beforeEach(() => {
        console.error = consoleErrorFn
        consoleErrorSpy = jest.spyOn(console, "error");
    });

    afterEach(() => {
        consoleErrorSpy.mockClear();
    });

    it("does not throw an error when props are provided", () => {
        shallow(<CardSubHeading {...props} />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(0);
    });

    it("throws an error when props are not of type `string`", () => {
        shallow(<CardSubHeading heading={0} />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(1);
    });

    it("throws an error when heading prop is not provided", () => {
        shallow(<CardSubHeading />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(1);
    });
});