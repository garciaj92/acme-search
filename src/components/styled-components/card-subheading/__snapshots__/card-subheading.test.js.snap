// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSubHeading: Test visual of component renders correctly 1`] = `
<h5
  className="text-muted mb-0"
>
  Test Heading
</h5>
`;
